sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveQl.scala

```diff
@@ -240,12 +242,24 @@ private[hive] object HiveQl {
      * Otherwise, there will be Null pointer exception,
      * when retrieving properties form HiveConf.
      */
-    val hContext = new Context(new HiveConf())
+    val hContext = new Context(hiveConf)
     val node = ParseUtils.findRootNonNullToken((new ParseDriver).parse(sql, hContext))
     hContext.clear()
     node
   }
 
+  /**
+   * Returns the HiveConf
+   * TODO get it from HiveContext?
+   */
+  private[this] def hiveConf(): HiveConf = {
```

Comment:
That sounds good to me.
On May 9, 2015 8:13 PM, "nitin2goyal" <notifications@github.com> wrote:

> In sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveQl.scala
> <https://github.com/apache/spark/pull/5963#discussion_r29998407>:
>
> >      val node = ParseUtils.findRootNonNullToken((new ParseDriver).parse(sql, hContext))
> >      hContext.clear()
> >      node
> >    }
> >
> > +  /**
> > +   * Returns the HiveConf
> > +   * TODO get it from HiveContext?
> > +   */
> > +  private[this] def hiveConf(): HiveConf = {
>
> Seems fine to me. Since it's already been done here, should I do just this
> change for 1.2 and 1.3 branches ?
> Issue - https://issues.apache.org/jira/browse/SPARK-7331
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/apache/spark/pull/5963/files#r29998407>.
>